glean-core/src/database/mod.rs

```diff
@@ -351,6 +423,14 @@ impl Database {
     ///
     /// * This function will **not** panic on database errors.
     pub fn clear_ping_lifetime_storage(&self, storage_name: &str) -> Result<()> {
+        let ping_lifetime_data = &self.ping_lifetime_data;
+        if let Some(ping_lifetime_data) = ping_lifetime_data {
+            ping_lifetime_data
+                .write()
+                .expect("Can't access ping lifetime data as writable")
+                .clear();
+        }
```

Comment:
Shouldn't we be returning here?

Reply:
My thinking was that we might have something to clear on the map **and** in rkv. I don't implement this in this PR yet, but will do it right after... What do you think?

Reply:
If the implementation that makes use of this code will come in a separate PR, it will make more sense to have this code as part of that future PR. Otherwise it's hard to test and to make sense of it and, more importantly, if we need to backout the PR/features it gets a bit complicated as the PR itself is not "self-contained" :)

Reply:
Makes sense, I will return for now then.